hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/PrecompileArgumentsDecoder.java

```diff
@@ -0,0 +1,334 @@
+package com.hedera.services.store.contracts.precompile;
+
+import com.hedera.services.utils.EntityIdUtils;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.TokenID;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.web3j.abi.FunctionReturnDecoder;
+import org.web3j.abi.TypeReference;
+import org.web3j.abi.datatypes.StaticArray;
+import org.web3j.abi.datatypes.Type;
+import org.web3j.abi.datatypes.generated.Bytes32;
+import org.web3j.abi.datatypes.generated.Int64;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class PrecompileArgumentsDecoder {
+
```

Comment:
Rather than hand writing the decoding I think we should use a library.  

https://github.com/esaulpaugh/headlong is the best one I've seen so far.  And it has a CLI - https://github.com/esaulpaugh/headlong-cli - that I used to validate ABI2 compatability.

```
java -jar headlong-cli-0.9-SNAPSHOT.jar -d "((address,(address,int64)[],(address,address,int64)[])[])" 00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000005432100000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000fedcba00000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000140000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000000000140000000000000000000000000000000000000000000000000000000000000578000000000000000000000000000000000000000000000000000000000000001500000000000000000000000000000000000000000000000000000000000005dc000000000000000000000000000000000000000000000000000000000000001600000000000000000000000000000000000000000000000000000000000006400000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000001700000000000000000000000000000000000000000000000000000000000000270000000000000000000000000000000000000000000000000000000000000025000000000000000000000000000000000000000000000000000000000000001800000000000000000000000000000000000000000000000000000000000000280000000000000000000000000000000000000000000000000000000000000026
(
  [ [ '054321', [  ], [  ] ], [ 'fedcba', [ [ '14', '0578' ], [ '15', '05dc' ], [ '16', '0640' ] ], [ [ '17', '27', '25' ], [ '18', '28', '26' ] ] ] ]
)
```